lrpar/src/lib/ctbuilder.rs

```diff
@@ -401,9 +401,8 @@ where
             .as_ref()
             .expect("output_path must be specified before processing.");
         let yk = match self.yacckind.clone() {
-            Some(YaccKind::SelfDescribing(_)) | None => {
-                panic!("yacckind must be specified before processing.")
-            }
+            None => YaccKind::SelfDescribing(None),
+            Some(YaccKind::SelfDescribing(default)) => YaccKind::SelfDescribing(default),
```

Comment:
This can be simplified to the following, but I suppose the current code is more robust to not potentially doing the wrong thing when adding new variants?

```
             None => YaccKind::SelfDescribing(None),
             Some(YaccKind::Eco) => panic!("Eco compile-time grammar generation not supported."),
             Some(x) => x,
```

So I didn't want to just simplify it without asking first.

Reply:
I think the simplification is the right thing to do here.

[TBH, we should drop the `Eco` support at some point. That tool has bitrotted, and carrying stuff around for it probably isn't worthwhile anymore.]